src/connector/src/source/cdc/enumerator/mod.rs

```diff
@@ -271,6 +272,140 @@ impl<T: CdcSourceTypeTrait> CdcMonitor for DebeziumSplitEnumerator<T> {
+                    }
+                }
+                if let Some((newest_file, newest_size)) = binlog_files.last() {
+                    // Extract sequence number from filename
+                    if let Some(seq) = Self::extract_binlog_seq(newest_file) {
+                        self.metrics
+                            .mysql_cdc_binlog_file_seq_max
+                            .with_guarded_label_values(&[hostname, port])
+                            .set(seq as i64);
+                        tracing::debug!(
+                            "MySQL CDC source {} ({}:{}): newest binlog = {}, seq = {}, size = {}",
+                            self.source_id,
+                            hostname,
+                            port,
+                            newest_file,
+                            seq,
+                            newest_size
+                        );
+                    }
+                }
+                tracing::debug!(
+                    "MySQL CDC source {} ({}:{}): total {} binlog files",
+                    self.source_id,
+                    hostname,
+                    port,
+                    binlog_files.len()
+                );
+            }
+            Err(e) => {
+                tracing::error!(
+                    "Failed to query binlog files for MySQL CDC source {} ({}:{}): {}",
+                    self.source_id,
+                    hostname,
+                    port,
+                    e.as_report()
+                );
+            }
+        }
+        Ok(())
+    }
+
+    /// Extract sequence number from binlog filename
+    /// e.g., "binlog.000001" -> Some(1), "mysql-bin.000123" -> Some(123)
+    fn extract_binlog_seq(filename: &str) -> Option<u64> {
+        filename.rsplit('.').next()?.parse::<u64>().ok()
+    }
+
+    /// Query binlog files from MySQL, returns Vec<(filename, size)>
+    async fn query_binlog_files(&self) -> ConnectorResult<Vec<(String, u64)>> {
+        // Extract connection parameters from CDC properties
+        let hostname = self
+            .properties
+            .get("hostname")
+            .ok_or_else(|| anyhow::anyhow!("hostname not found in CDC properties"))?;
+        let port = self
+            .properties
+            .get("port")
+            .ok_or_else(|| anyhow::anyhow!("port not found in CDC properties"))?;
+        let user = self
+            .properties
+            .get("username")
+            .ok_or_else(|| anyhow::anyhow!("username not found in CDC properties"))?;
+        let password = self
+            .properties
+            .get("password")
+            .ok_or_else(|| anyhow::anyhow!("password not found in CDC properties"))?;
+        let database = self
+            .properties
+            .get("database.name")
+            .ok_or_else(|| anyhow::anyhow!("database.name not found in CDC properties"))?;
+
+        // Build MySQL connection string
+        let mysql_url = format!(
+            "mysql://{}:{}@{}:{}/{}",
```

Comment:
How about the SSL related configs? https://github.com/risingwavelabs/risingwave/blob/d052368d0f54631b5649dce648b70ac323634954/src/connector/src/source/cdc/external/mysql.rs#L448

Can we extract the common codes to set up mysql connection based on the with properties? If I search for `mysql_async::Pool::new` in our code bases, there are many occurrences and I think it is easy to miss some configs if we don't unify them.

Reply:
Btw, I found that we hard code ssl mode for PG rather than honoring the user provided option when querying the LSN. Will this be an issue?
https://github.com/risingwavelabs/risingwave/blob/d052368d0f54631b5649dce648b70ac323634954/src/connector/src/source/cdc/enumerator/mod.rs#L211

Reply:
> in our code bases, there are many occurrences and I think it is easy to miss some configs if we don't unify them.

There are many places in our code bases using this connection pool in many places:
- mysql cdc backfill
- mysql tvf
- new StarRocks schema client
- this pr, monitoring upstream mysql binlog

I plan to extract a common method like `build_mysql_connection_pool`  first to unify the CDC backfill and monitoring, and then unify the rest in later separate pr. This will prevent the changes in this pr from being too broad.